WcaOnRails/app/models/competition.rb

```diff
@@ -41,22 +41,27 @@ class Competition < ActiveRecord::Base
 
   # We have stricter validations for confirming a competition
   [:cityName, :countryId, :venue, :venueAddress, :website, :latitude, :longitude].each do |field|
-    validates field, presence: true, if: :isConfirmed?
+    validates field, presence: true, if: :confirmedOrVisible?
   end
-  validate :must_have_at_least_one_event, if: :isConfirmed?
+
+  validate :must_have_at_least_one_event, if: :confirmedOrVisible?
   def must_have_at_least_one_event
     if events.length == 0
       errors.add(:eventSpecs, "Competition must have at least one event")
     end
   end
 
-  validate :must_have_at_least_one_delegate, if: :isConfirmed?
+  validate :must_have_at_least_one_delegate, if: :confirmedOrVisible?
   def must_have_at_least_one_delegate
     if delegate_ids.length == 0
       errors.add(:delegate_ids, "Competition must have at least one WCA delegate")
     end
   end
 
+  def confirmedOrVisible?
```

Comment:
Please use snake case here =) These `isConfirmed` and `showAtAll` are camelCased because the tables was created and used originally in PHP code.